src/services/vector_store/pinecone_store.py

```
@@ -0,0 +1,139 @@
+                )
+
+            # Upsert in batches
+            batch_size = 100
+            for i in range(0, len(pinecone_vectors), batch_size):
+                batch = pinecone_vectors[i : i + batch_size]
+                self.index.upsert(vectors=batch)
+
+            print(f"✅ Added {len(vectors)} vectors to Pinecone")
+
+        except Exception as e:
+            raise ConfigurationError(f"Failed to add vectors to Pinecone: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)